### PR TITLE
Feature: UI for Provider Profile

### LINKED ISSUE #201: Match Provider profile management UI Screen to Figma Screen
- Create Provider Profile management UI screen to match sigma screen
- Required:
   - New UI should look like the Figma Screen
   - If Some Data is missing from the Profile class don't modify it, put placeholders in the UI
   - To be done when Figma is done

### LINKED ISSUE #204: Complete the Provider VM
Required: 
- Complete the provider VM to implement profile management
- Adjust Data class & Repository if needed
- Use it in the provider profile management UI

### PR DESCRIPTION
**Pull request by ethanboren**
### Description:
- Created the user interface that allows the provider to view their profile.
- Connected the UI with the provider viewModel
- Performed tests to ensure everything works correctly.

### Related Issues:
- Fixes #201
- Fixes #204

### Tests Performed:
- Verified that the provider's profile correctly displays data from the database.
- Successfully ran unit and integration tests.

### Todo :
- Integrate completed jobs (`JobsDoneSection()`) into the UI (to replace hardcoded data)
- Update the Figma design to reflect recent changes

### CODE REVIEW

File: app/src/main/java/com/android/solvit/MainActivity.kt
Comment (yshdb5) [inline app/src/main/java/com/android/solvit/MainActivity.kt:1]: Consider maybe checking the current user with `Firebase.auth.currentUser?.uid` directly in `ProfessionalProfile.kt `

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt
Comment (yshdb5) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt:1]: **A few comments:**

- The `JobsDoneSection` section is hardcoded, we did not implement the logic to retrieve the jobs done but they  4 jobs items should be replaced later by the correct iteration on the accomplished jobs list associated to the provider.
- The Position availability button is not updating anything outside of the provider UI. It should be connected someway to all the seekers to announce the availability of the provider or it should be developed a bit to enable the provider to do something interesting with it.
- Some fields in the bottom part of the UI are also hardcoded like the "Jobs Completed" and some are useless for now because they do not correspond to any feature implemented or that will be implemented in the near future (eg. pay range or availability).

File: app/src/main/res/drawable/professionalprofilelogo.png
Comment (yshdb5) [inline app/src/main/res/drawable/professionalprofilelogo.png:1]: I think that this image is never used, it could be deleted.

**Review by yshdb5 — APPROVED** [review]
### Overall Comment
This PR implements a new UI screen for the Provider's profile and integrates the Provider ViewModel with the UI for dynamic updates. The changes improve both the functionality and the test coverage of the application, ensuring a more robust and user-friendly experience.

### Strong Elements
- **Dynamic UI Updates**: Connecting the ViewModel to the UI for dynamic updates based on the logged-in user enhances the app's responsiveness and user experience.
- **Clear Code Structure**: The code is well-structured and follows consistent formatting, making it easy to read and understand.

### Areas for Improvement
- **Comments**: I have written some comments on some files with remarks and suggestions.
- **Test Coverage**: While the tests are comprehensive, adding more edge case scenarios could further strengthen the test coverage.
- **Documentation**: Including comments or documentation for complex logic or less obvious parts of the code would enhance maintainability and help future developers understand the code better.

### Conclusion
Overall, this PR significantly improves the application's functionality. With additional edge cases in tests, and enhanced documentation, the changes will be even more robust and maintainable. Goof job on these additions!

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+import com.google.firebase.Firebase
+import com.google.firebase.auth.auth
+
+@Composable
+fun ProfessionalProfileScreen(
+    listProviderViewModel: ListProviderViewModel =
+        viewModel(factory = ListProviderViewModel.Factory),
+    navigationActions: NavigationActions
+) {
+  val userId = Firebase.auth.currentUser?.uid ?: "-1"
```
Comment (achillet3) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt:58]: Nit: Instead of querying once again to Firebase the currently authenticated user, you could pass as an argument the AuthViewModel which holds the uid of the current user.

File: app/src/androidTest/java/com/android/solvit/provider/ui/profile/ProfessionalProfileTest.kt
Comment (achillet3) [inline app/src/androidTest/java/com/android/solvit/provider/ui/profile/ProfessionalProfileTest.kt:1]: Good tests with good coverage. 
# Potential Improvements
- Verify that the first two calls to the VM work correctly (more details below)
- Check that the "background" column is correctly displayed. 
You could look at `ListProviderScreenTest.kt` to see how to test the VM calls.

File: app/src/main/java/com/android/solvit/MainActivity.kt [hunk 2]
```diff
@@ -148,6 +150,7 @@ fun SeekerUI(
   }
 }
 
+@SuppressLint("StateFlowValueCalledInComposition")
```
Comment (achillet3) [inline app/src/main/java/com/android/solvit/MainActivity.kt:153]: I tried without this tag and the code works fine. What is its purpose?

File: app/src/main/java/com/android/solvit/MainActivity.kt [hunk 3]
```diff
@@ -163,5 +166,9 @@ fun ProviderUI(
     }
     composable(Route.MAP_OF_SEEKERS) { ProviderMapScreen(navigationActions = navigationActions) }
     composable(Screen.CALENDAR) { ProviderCalendarScreen(navigationActions = navigationActions) }
+    composable(Screen.PROFESSIONAL_PROFILE) {
+      ProfessionalProfileScreen(
+          listProviderViewModel = listProviderViewModel, navigationActions = navigationActions)
```
Comment (achillet3) [inline app/src/main/java/com/android/solvit/MainActivity.kt:171]: Nit: Simply use the name of the view models directly to improve readability

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+  val provider =
+      listProviderViewModel.providersList.collectAsState().value.first { it.uid == userId }
+  Column(modifier = Modifier.fillMaxSize().background(Color.White).testTag("background")) {
```
Comment (achillet3) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt:61]: Nit: The test tag doesn't accurately describe this column, which isn't a background but the screen in itself

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt [hunk 5]
```diff
@@ -0,0 +1,324 @@
...
+    ProfileHeader(navigationActions = navigationActions, provider = provider)
```
Comment (achillet3) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt:62]: Nit: You could pass the variables directly without matching them (i.e., `navigationActions` instead of `navigationActions = navigationActions`. The names of the variables are explicit enough that we understand what is passed as what.

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt [hunk 6]
```diff
@@ -0,0 +1,324 @@
...
+                      onClick = { navigationActions.navigateTo(Route.REQUESTS_FEED) },
+                      modifier = Modifier.testTag("backButton")) {
+                        Icon(
+                            Icons.AutoMirrored.Filled.ArrowBack,
+                            contentDescription = "Back",
+                            modifier = Modifier.size(24.dp),
+                            tint = Color(239, 70, 55))
+                      }
+                }
+              }
```
Comment (achillet3) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt:93]: Nit: I understand that this is done to follow the Figma, but it is a little laborious, and I would argue that the gain in visual design does not outweigh the loss of code clarity. I suggest creating an explicit top bar with the back arrow, the screen name, and maybe a menu icon to access other features in the future.

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt [hunk 7]
```diff
@@ -0,0 +1,324 @@
...
+            text = title,
+            fontSize = 14.sp,
+            textAlign = TextAlign.Center,
+            maxLines = 2,
+            color = Color(239, 70, 55))
+      }
+}
+
+@Composable
+fun StatsSection(provider: Provider) {
```
Comment (achillet3) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt:253]: Nit: The columns are not aligned in the UI

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt
Comment (achillet3) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt:1]: # Overall Comments
Well done implementing this treacherous UI design. I overall don't feel the Figma design is very readable and clear and unfortunately this is emphasized when coding this complex screen. The code could benefit from being a little bit better structured and using better the already existing components of the app.

# Potential Improvements
- Standardize the page to improve code readability and overall screen uniformity
- Make sure the colors correspond to the app's UI color scheme (this will be addressed in a future PR of mine)
- Make the screen clearer and easier to read and understand. I'm not sure this screen portrays the most important information of a Provider's Profile

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt [hunk 8]
```diff
@@ -0,0 +1,324 @@
...
+import com.android.solvit.seeker.model.provider.ListProviderViewModel
+import com.android.solvit.shared.model.provider.Provider
+import com.android.solvit.shared.ui.authentication.VerticalSpacer
+import com.android.solvit.shared.ui.navigation.NavigationActions
+import com.android.solvit.shared.ui.navigation.Route
+import com.google.firebase.Firebase
+import com.google.firebase.auth.auth
+
+@Composable
+fun ProfessionalProfileScreen(
```
Comment (achillet3) [inline app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt:53]: Nit: Rename the screen to ProviderProfileScreen to keep naming consistency across the app

File: app/src/main/res/drawable/professionalprofilelogo.png
Comment (achillet3) [reply app/src/main/res/drawable/professionalprofilelogo.png:1]: +1

**Review by achillet3 — COMMENTED** [review]
# Overall Comments
- Good effort implementing this complex screen. I would prefer spending some time redesigning the Figma Screen to make sure that the screen has all the important info (especially at this point). Maybe making it a little closer to the simple Seeker Profile would be more helpful and better at this stage.
- Don't forget to use the components already developed to prevent unnecessary queries to Firebase for example. 
- Keep the naming convention as consistent as possible to help future-proofing the code

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt
Comment (ethanboren) [reply app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt:1]: Yes I know, I noted it in the description of the pull request and I added todo in the code

Comment (ethanboren) [reply app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt:1]: Regarding the figma and clarity of design, it was accepted at meetings and decided to integrate UI into code. It is not nice to comment on this topic after having coded it and not during the meeting. Regarding the next comments, I will make the changes

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt (hunk 1 above)
Comment (ethanboren) [reply app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt:58]: I decided to do it like this in order to less overload the main activity and do as Yshai and Hassan did in order to keep a certain consistency

File: app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt (hunk 7 above)
Comment (ethanboren) [reply app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt:253]: I know, it’s because some lines have 2 information and another 3. Anyway when we have all the information, I will update this screen

Comment (ethanboren) [reply app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt:253]: I know, it’s because some lines have 2 information and another 3. Anyway when we have all the information, I will update this screen

Comment (achillet3) [reply app/src/main/java/com/android/solvit/provider/ui/profile/ProfessionalProfile.kt:253]: I was referring to the first 2 rows with both 2 elements. But I agree this is a nitpick and should be addressed when the UI is able to access all the necessary information

**Review by achillet3 — COMMENTED** [review]
All the changes look good to me! Just a final nit then I will approve: please change the name of the files to match the new composable functions - i.e, `ProviderProfile.kt` instead of `ProfessionalProfile.kt`

**Review by achillet3 — APPROVED** [review]
LGTM!
